src/transports/TransportEnvelope.js

```diff
@@ -0,0 +1,106 @@
+import AssertionFailed from '@gdbots/pbj/exceptions/AssertionFailed';
+import JsonSerializer from '@gdbots/pbj/serializers/JsonSerializer';
+
+const serializers = new Map();
+
+export default class TransportEnvelope {
+  /**
+   * @param {Message} message
+   * @param {string} serializer
+   */
+  constructor(message, serializer) {
+    Object.defineProperty(this, 'message', { value: message });
+    Object.defineProperty(this, 'serializer', { value: serializer });
+  }
+
+  /**
+   * @param {string} serializer
+   *
+   * @returns {JsonSerializer}
+   */
+  static getSerializer(serializer) {
+    if (!serializers.has(serializer)) {
+      serializers.set(serializer, JsonSerializer);
+    }
+
+    return serializers.get(serializer);
+  }
+
+  /**
+   * Recreates the envelope from a json string.
+   *
+   * @param {string} json
+   *
+   * @returns {TransportEnvelope}
+   *
+   * @throws {AssertionFailed}
+   */
+  static fromJSON(json) {
+    let obj;
+
+    try {
+      obj = JSON.parse(json);
+    } catch (e) {
+      throw new AssertionFailed('Invalid JSON.');
+    }
+
+    const serializer = obj.serializer ? obj.serializer : 'json';
+    const message = TransportEnvelope.getSerializer(serializer).deserialize(obj.message ? obj.message : '');
+
+    return new this(message, serializer);
+  }
+
+  /**
+   * @param {Object} obj
+   *
+   * @returns {TransportEnvelope}
+   */
+  static fromObject(obj = {}) {
```

Comment:
@albertgcsula fromObject needs to address the `is_replay` boolean.  that must be restored otherwise we lose that capability.